### ISSUE TITLE
Radial control is not recognized in editor window moved out of main window

### ISSUE THREAD

**Issue by xperiandri** [madskristensen/DialToolsForVS#18]
### Installed product versions
- Visual Studio: 15.5.2 Enterprise
- This extension: 0.7.46

### Description
If you have [Double-Click Maximize](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.Double-ClickMaximize) and maximize editor with double click then radial control do not function in it (no menu items appear).

### Steps to recreate
1. Drag editor outside the main Visual Studio window
2. Hold radial control to display it's menu

### Current behavior
No Visual Studio specific items appear

### Expected behavior
Visual Studio specific menu items appear and function the same as in the main window

**Comment by xperiandri** [comment]
@madskristensen how can I determine if window contains text editor? So that I cerate another instance of ControllerHost for that window.

**Comment by madskristensen** [comment]
@xperiandri Can you check `dte.ActiveDocument`?

**Comment by xperiandri** [comment]
I tried to do this
``` C#
    var dteEvents = (EnvDTE80.Events2)dte.Events;
    dteEvents.WindowEvents.WindowCreated += OnWindowCreated;
    dteEvents.WindowEvents.WindowClosing += OnWindowClosing;
    compositionService.DefaultCompositionService.SatisfyImportsOnce(this);
}

private static void OnWindowCreated(EnvDTE.Window window7)
{
    var window8 = (EnvDTE80.Window2)window7;

    controllersMapping.Add(window8, DialControllerHost.Instance.CreateController(window8));
}

private static void OnWindowClosing(EnvDTE.Window window)
{
    if (controllersMapping.TryGetValue(window, out RadialController controller))
    {
        DialControllerHost.Instance.UnsubscribeFromController(controller);
        controllersMapping.Remove(window);
    }
}
```

But it does not trigger when new WPF window is created by Visual Studio which I need to determine

**Comment by xperiandri** [comment]
It is because I need not DTE Windows but WPF Window of Visual Studio

**Comment by xperiandri** [comment]
But my PR does not have a fix to this issue, only a foundation to fix it.
And I have an idea how to determine WPF window creation.
But I'm open to suggestions.

**Comment by miloush** [comment]
Can we re-open this issue? I hit this too in 15.9.0 P3. It's breaking the Navigation experience if it swaps to a detached window during rotating.

**Comment by CobraCalle** [comment]
The problem occurs even when visual studio IS on the main monitor but other tool windows are to other monitors...

**Comment by xperiandri** [comment]
@CobraCalle would you like to fix?
I have an idea and I can explain what to do.
I just can't spend time on it now unfortunately ☹

**Comment by xperiandri** [comment]
@madskristensen I thought that we need to create a radial controller per FloatingWindow (Visual Studio derived class from WPF WIndow).
@olegkrawchuk successfully prototyped this idea with Win API hook but the controller does not appear in a floating window.
Do you have any ideas which HWND we need to create a controller based on?
https://github.com/olegkrawchuk/DialToolsForVS/tree/issues/18

**Comment by xperiandri** [comment]
Looks like we have the right HWND according to https://github.com/kusanagi2k2/SurfaceDialWPFDemo/blob/b43e864b5761205e1477f9aa1f3fe937b005929e/SurfaceDialExample/MainWindow.xaml.cs
But in our case, it does not work for the child window.
Any ideas?

**Comment by xperiandri** [comment]
@kusanagi2k2, maybe you know what can be wrong?

**Comment by xperiandri** [comment]
Fixed in #41

**Comment by xperiandri** [comment]
I've just missed that new controllers are not filled with menu items
